source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -118,41 +119,30 @@ generated by oVirt:
 
 ![](../../../../images/wiki/upload-ova.png)
 
-1. The front end retrieves the OVF configuration from the OVA.
-2. The front end sends the OVF configuration to the back end.
-3. The back end parses the OVF configuration and returns a VM to the
-   front end.
-4. The front end sends (optionally customized) parameters of the import
-   process to the back end.
-5. The back end persists the VM into the database.
-6. The back end triggers the creation of the target VM disks on the
-   storage domain.
-7. The back end sends the import parameters - including the destination
-   disks - to imageio-proxy, which sends them to imageio.
-8. Imageio-proxy sends the engine a URL to write to. This URL is sent to
-   the front end.
-9. The front end streams the **whole OVA** to imageio-proxy, which
-   streams it to imageio.
-10. Imageio reads the data and according to the import parameters
-    streams the data into the corresponding target disks.
+1. The frontend retrieves the OVF configuration from the OVA.
+2. The frontend sends the OVF configuration to the backend.
+3. The backend parses the OVF configuration and returns a VM object
+   to the frontend.
+4. The frontend sends back the VM (optionally customized) to the backend.
+5. The backend persists the VM into the database and lock it.
+6. The backend sends the list of disks metadata to the frontend,
+   which starts the process of uploading the disks.
```

Comment:
No, just the list of disks to upload, the the metadata need to extract the disks from the ova.

The actual disks are created (and attached to the vm) when uploading a disk. This way if the operation fail we need to clean up only the disk that were uploaded, the disk we did not upload yet do not exit yet.

If we want to create all the disks when creating the vm, we need to the change the image upload api to upload to existing disk, which is not supported now. If there is a good reason to do this, why not.